src/garage/torch/policies/discrete_qf_derived_policy.py

```diff
@@ -0,0 +1,72 @@
+"""A Discrete QFunction-derived policy.
+
+This policy chooses the action that yields to the largest Q-value.
+"""
+import torch
+
+from garage.torch.policies.policy import Policy
+
+
+class DiscreteQFDerivedPolicy(Policy):
+    """Policy that derives its actions from a learned Q function.
+
+    The action returned is the one that yields the highest Q value for
+    a given state, as determined by the supplied Q function.
+
+    Args:
+        qf (object): Q network.
+        env_spec (EnvSpec): Environment specification.
+        name (str): Name of this policy.
+    """
+
+    def __init__(self, qf, env_spec, name='DiscreteQFDerivedPolicy'):
```

Comment:
i don't think name is needed for torch?

Reply:
`garage.torch.policy` requires one, though it doesn't look like it's being used anywhere so not sure why.